src/Core/Services/CryptoService.cs

```diff
@@ -157,10 +163,26 @@ public async Task<UserKey> DecryptUserKeyWithMasterKeyAsync(MasterKey masterKey,
             if (encUserKey == null)
             {
                 var userKeyMasterKey = await _stateService.GetMasterKeyEncryptedUserKeyAsync(userId);
-                if (userKeyMasterKey == null)
+
+                if (userKeyMasterKey is null)
                 {
-                    throw new Exception("No encrypted user key found");
+                    // Migrate old key
+                    var oldEncUserKey = await _stateService.GetEncKeyEncryptedAsync(userId);
+
+                    if (oldEncUserKey is null)
+                    {
+                        throw new Exception("No encrypted user key nor old encKeyEncrypted found");
+                    }
+
+                    var userKey = await DecryptUserKeyWithMasterKeyAsync(
+                        masterKey,
+                        new EncString(oldEncUserKey),
+                        userId
+                    );
+                    await SetMasterKeyEncryptedUserKeyAsync(oldEncUserKey, userId);
+                    return userKey;
```

Comment:
🤔  @jlf0dev I may have forgotten here to set the `encKey` to `null` after the migration succeeds. What do you think? Shall we maintain the `encKey` just in case, or just reset it before `return userKey`?